src/components/DetailsModal.tsx

```diff
@@ -96,76 +97,42 @@ const capitaliseWords = (str: string) => {
     .join(" ");
 };
 
-export const DetailsModal: FunctionComponent<DetailsModalProps> = ({
+const DetailsModal: FunctionComponent<DetailsModalProps> = ({
   emotion,
   onClose,
+  session,
 }): ReactElement => {
-  const [mood, setMood] = useState<Record<string, any>>(initialFormState);
-  const link = useRef<HTMLAnchorElement>(null);
+  const [mood, setMood] = useState(initialFormState);
   const [error, setError] = useState("");
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const link = useRef<HTMLAnchorElement>(null);
   const router = useRouter();
 
-  //could be refactored again with Mark's updates
-  const createOrUpdateEntry = async (existingEntry: any, updatedMood: any) => {
-    if (existingEntry.data) {
-      const { data: updatedEntry, error } = await supabaseBrowser
-        .from("entries")
-        .update({
-          mood: updatedMood.mood,
-          journal_entry: updatedMood.journal_entry,
-          context_people: updatedMood.context_people,
-          context_location: updatedMood.context_location,
-        })
-        .eq("id", existingEntry.data.id);
-
-      return { data: updatedEntry, error };
-    } else {
-      return await supabaseBrowser.from("entries").insert(updatedMood);
-    }
-  };
-
   const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-
-    const {
-      data: { session },
-    } = await supabaseBrowser.auth.getSession();
-
-    if (!session) return;
-
     const user = session.user;
-    const updatedMood: Mood = {
+
+    setMood({
       ...mood,
       user_id: user.id,
       mood: emotion,
-      mood_date: mood.mood_date,
-      journal_entry: mood.journal_entry,
-      context_people: mood.context_people,
-      context_location: mood.context_location,
-    };
-    setMood(updatedMood);
-
-    const existingEntry = await supabaseBrowser
-      .from("entries")
-      .select("*")
-      .eq("mood_date", updatedMood.mood_date)
-      .eq("user_id", updatedMood.user_id)
-      .single();
+    });
+    setHasSubmitted(true);
```

Comment:
Type error: Type 'string' is not assignable to type 'null'.
--
17:06:06.019 |  
17:06:06.019 | 116 \|       ...mood,
17:06:06.019 | 117 \|       user_id: user.id,
17:06:06.019 | > 118 \|       mood: emotion,
17:06:06.019 | \|       ^
17:06:06.019 | 119 \|     });
17:06:06.019 | 120 \|     setHasSubmitted(true);
17:06:06.019 | 121 \|   };